optimum/exporters/tasks.py

```diff
@@ -201,6 +203,7 @@ class TasksManager:
         "stable-diffusion": "diffusers",
         "zero-shot-image-classification": "transformers",
         "zero-shot-object-detection": "transformers",
+        "timm-image-classification": "timm",
```

Comment:
Well, models for the very same task (aka pipeline tag) may rely on a different library:
* task `image-classification`, library `timm`: https://huggingface.co/timm/resnetaa101d.sw_in12k
* task `image-classification`, library `transformers`: https://huggingface.co/microsoft/resnet-50

Given that we have a way to distinguish whether the model repo is from timm or transformers automatically, I don't see the need for the `_TASKS_TO_LIBRARY` map, nor for adding `image-classification-timm` (which is not a task, but a couple task, library).

But currently we have `_TASKS_TO_AUTOMODELS` that maps one to one. This would not be the case anymore, so either we could have it as `Dict[str, Union[str, Tuple[str]]]` like I did in https://github.com/huggingface/optimum/pull/967 with `"automatic-speech-recognition": ("AutoModelForSpeechSeq2Seq", "AutoModelForCTC")`, delaying the complexity in later functions, or we would need an additional map library, task -> loading class that would allow to distinguish (in the case of transformers/timm, but unfortunately not in the ASR case, I rather rely on the `_model_mapping` from automodel).

WDYT @mht-sharma ?